corehq/apps/app_manager/util.py

```diff
@@ -134,9 +134,11 @@ def change_xmlns(xform, old_xmlns, new_xmlns):
     else:
         GENERIC_XMLNS = "http://www.w3.org/2002/xforms"
         tag_xmlns = xform.data_node.tag_xmlns
+        form_xmlns = "http://openrosa.org/formdesigner/%s" % form.get_unique_id()
         if not tag_xmlns or tag_xmlns == GENERIC_XMLNS:  # no xmlns
-            xmlns = "http://openrosa.org/formdesigner/%s" % form.get_unique_id()
-            xml = change_xmlns(xform, GENERIC_XMLNS, xmlns)
+            xml = change_xmlns(xform, GENERIC_XMLNS, form_xmlns)
+        elif tag_xmlns != form_xmlns:
+            xml = change_xmlns(xform, tag_xmlns, form_xmlns)
```

Comment:
This is definitely going to break things. It looks like everytime an xform is saved it will check that the xmlns and form_unique_id are the same and if it isn't, it changes the xmlns. I checked icds and those two are not the same.